core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -49,17 +49,56 @@
  */
 @Extension
 public class ChannelPinger extends ComputerListener {
+    static final int PING_TIMEOUT_SECONDS_DEFAULT = 4 * 60;
+    static final int PING_INTERVAL_SECONDS_DEFAULT = 5 * 60;
+    
     private static final Logger LOGGER = Logger.getLogger(ChannelPinger.class.getName());
-    private static final String SYS_PROPERTY_NAME  = ChannelPinger.class.getName() + ".pingInterval";
-    private static final int DEFAULT_PING_INTERVAL_MIN = 5;
+    private static final String SYS_PROPERTY_NAME_TIMEOUT_SECONDS  = ChannelPinger.class.getName() + ".pingTimeoutSeconds";
+    private static final String SYS_PROPERTY_NAME_INTERVAL_MINUTES_DEPRECATED  = ChannelPinger.class.getName() + ".pingInterval";
+    private static final String SYS_PROPERTY_NAME_INTERVAL_SECONDS  = ChannelPinger.class.getName() + ".pingIntervalSeconds";
     
     /**
-     * Interval for the ping in minutes.
+     * Timeout for the ping in seconds.
+     */
+    private int pingTimeoutSeconds = PING_TIMEOUT_SECONDS_DEFAULT;
+
+    /**
+     * Interval for the ping in seconds.
      */
-    private final int pingInterval;
+    private int pingIntervalSeconds = PING_INTERVAL_SECONDS_DEFAULT;
 
     public ChannelPinger() {
-        pingInterval = SystemProperties.getInteger(SYS_PROPERTY_NAME, DEFAULT_PING_INTERVAL_MIN);
+        String interval = System.getProperty(SYS_PROPERTY_NAME_INTERVAL_SECONDS);
+        if (interval != null) {
+            try {
+                pingIntervalSeconds = Integer.parseInt(interval);
+            } catch (NumberFormatException e) {
+                LOGGER.warning("Ignoring invalid " + SYS_PROPERTY_NAME_INTERVAL_SECONDS + "=" + interval);
+            }
+        }
+        
+        // if interval wasn't set we read the deprecated property in minutes
+        if (interval == null) {
+            interval = System.getProperty(SYS_PROPERTY_NAME_INTERVAL_MINUTES_DEPRECATED);
+            if (interval != null) {
+                LOGGER.warning(SYS_PROPERTY_NAME_INTERVAL_MINUTES_DEPRECATED + " property is deprecated, " + SYS_PROPERTY_NAME_INTERVAL_SECONDS
+                        + " should be used");
+                try {
+                    pingIntervalSeconds = Integer.parseInt(interval) * 60;
+                } catch (NumberFormatException e) {
+                    LOGGER.warning("Ignoring invalid " + SYS_PROPERTY_NAME_INTERVAL_MINUTES_DEPRECATED + "=" + interval);
+                }
+            }
+        }
+        
+        String timeout = System.getProperty(SYS_PROPERTY_NAME_TIMEOUT_SECONDS);
+        if (timeout != null) {
+            try {
+                pingTimeoutSeconds = Integer.parseInt(timeout);
+            } catch (NumberFormatException e) {
+                LOGGER.warning("Ignoring invalid " + SYS_PROPERTY_NAME_TIMEOUT_SECONDS + "=" + timeout);
+            }
+        }
```

Comment:
Yes I'm going to fix that, it really needs to use `SystemProperties` instead of `System.getProperty` to be able to get the properties from the `ServletContext`but I also have to take into account the deprecated property, so is not going to be so lean.